assets/js/modules/analytics/common/profile-name.js

```diff
@@ -0,0 +1,83 @@
+/**
+ * Analytics Property Select component.
+ *
+ * Site Kit by Google, Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * WordPress dependencies
+ */
+import { useCallback } from '@wordpress/element';
+import { __ } from '@wordpress/i18n';
+
+/**
+ * Internal dependencies
+ */
+import Data from 'googlesitekit-data';
+import { STORE_NAME as CORE_FORMS } from '../../../googlesitekit/datastore/forms';
+import { TextField, HelperText, Input } from '../../../material-components';
+import { STORE_NAME, PROFILE_CREATE, FORM_SETUP } from '../datastore/constants';
+const { useSelect, useDispatch } = Data;
+
+export default function ProfileName() {
+	const propertyID = useSelect( ( select ) => select( STORE_NAME ).getPropertyID() );
+	const profiles = useSelect( ( select ) => select( STORE_NAME ).getProfiles( propertyID ) );
+	const profileID = useSelect( ( select ) => select( STORE_NAME ).getProfileID() );
+	const profileName = useSelect( ( select ) => select( CORE_FORMS ).getValue( FORM_SETUP, 'profileName' ) );
```

Comment:
Let's default this to `__( 'All Web Site Data', 'google-site-kit' )` as the field should be pre-filled with this when the user selects the option to "set up a new view". We'll probably need to use `useEffect` to set this in the form state on the fly.

This is a slightly better UX as well as the field label looks like a value when the field is empty.

I realize we have the default in place on the server side too; that's fine and we can keep that as the endpoint there should have that default as well - this is just about the UI.

Reply:
Updated.